gridcapa-cse-valid-publication-app/src/main/java/com/farao_community/farao/cse_valid_publication/app/CseValidPublicationService.java

```diff
@@ -9,104 +9,154 @@
-            List<TTimestamp> timestampsToBeValidated = tcDocument.getAdjustmentResults().get(0).getTimestamp();
-            LOGGER.info("TTC adjustment file contains {} timestamps to be validated", timestampsToBeValidated.size());
-            String cseValidRequestId = UUID.randomUUID().toString();
-            timestampsToBeValidated.forEach(ts -> timestampCseValidRequests.put(ts, buildCseValidRequest(process, cseValidRequestId, ttcAdjustmentFilePath, ts)));
-            Map<TTimestamp, CompletableFuture<CseValidResponse>> timestampCseValidResponses = new HashMap<>();
-            try {
-                runCseValidRequests(timestampCseValidRequests, timestampCseValidResponses);
-            } catch (InterruptedException e) {
-                LOGGER.error("Error during Cse valid running for date {}", targetDate, e);
-                Thread.currentThread().interrupt();
-                throw new CseValidPublicationInternalException(String.format("Error during Cse valid running for date %s ", targetDate), e);
-            } catch (ExecutionException e) {
-                LOGGER.error("Error during Cse valid running for date {}", targetDate, e);
-                throw new CseValidPublicationInternalException(String.format("Error during Cse valid running for date %s ", targetDate), e);
+        LOGGER.info("Target date with offset: {}", targetDateWithOffset);
+
+        String processCode = ProcessUtils.getProcessCode(process);
+        TcDocumentTypeWriter tcDocumentTypeWriter = new TcDocumentTypeWriter(processCode, targetDateWithOffset);
+
+        String requestUrl = urlConfiguration.getTaskManagerBusinessDateUrl() + targetDateWithOffset;
+        LOGGER.info("Requesting URL: {}", requestUrl);
+        ResponseEntity<TaskDto[]> responseEntity = restTemplateBuilder.build().getForEntity(requestUrl, TaskDto[].class);
+
+        TaskDto[] taskDtoArray = responseEntity.getBody();
+        if (taskDtoArray != null && taskDtoArray.length > 0) {
+            Optional<ProcessFileDto> ttcAdjustmentFileDtoOpt = getProcessFile(taskDtoArray[0], "TTC_ADJUSTMENT");
+
+            if (ttcAdjustmentFileDtoOpt.isPresent()) {
+                TcDocumentType tcDocument = fileImporter.importTtcFile(ttcAdjustmentFileDtoOpt.get().getFileUrl(), "adjustment");
+
+                if (tcDocument != null) {
+                    validateTtc(process, initialTargetDate, targetDateWithOffset, taskDtoArray, tcDocument, tcDocumentTypeWriter);
+                    return;
+                }
             }
-            fillResultForAllTimestamps(timestampCseValidResponses);
-        } else {
-            tcDocumentTypeWriter.fillWithNoTtcAdjustmentError();
         }
-        fileExporter.saveTtcValidation(tcDocumentTypeWriter, process, localTargetDate);
-        return true;
+
+        tcDocumentTypeWriter.fillWithNoTtcAdjustmentError();
+        fileExporter.saveTtcValidation(tcDocumentTypeWriter, process, targetDateWithOffset);
```

Comment:
Can we do it more straightforward here ? Because in the end the final step is to save TTC validation file